### DISCUSSION TITLE
Adonis Session

### DISCUSSION THREAD

**Discussion by rust142** [adonisjs/core#4544] [Help] [answered]
![image](https://github.com/adonisjs/core/assets/56282560/adfea5a7-eefb-45ce-bf6b-99367bb90599)

I'm creating a session, and following the directions from the documentation.. but why is it like this.

LoginContoller: 
```ts
import User from "#models/user";

async submitForm(ctx: HttpContext) {
  let email = ctx.request.input("email");
  let password = ctx.request.input("password");

  const user = await User.verifyCredentials(email, password);

  await ctx.auth.use("web").login(user); // <-- Error

  return ctx.response.redirect().toRoute("dashboard_core.index");
}
```
Config/auth:
```ts
const authConfig = defineConfig({
  default: "web",
  guards: {
    web: sessionGuard({
      useRememberMeTokens: false,
      provider: sessionUserProvider({
        model: () => import("#models/user"),
      }),
    }),
  },
});
```

User verify credentials:

![image](https://github.com/adonisjs/core/assets/56282560/8b1ad224-bd58-4016-970b-0d1dbb6baf29)

**Comment by rust142** [comment]
This error occurs  after I clear the cookie that binds the session

**Comment by rust142** [comment]
When I stopped the server and changed to another port or use `node ace generate:key` it worked without any problems as above

**Comment by thetutlage** [comment]
The issue comes in HMR mode and we can reproduce it. Will come back with more information soon.

**Comment by Julien-R44** [comment, marked as answer]
That was indeed a problem with Hot Hook. Please updgrade to `hot-hook@0.2.2` and the problem should be fixed. Let me know if it's ok for you
